pfcpiface/errors.go

```diff
@@ -5,22 +5,85 @@ package main
 import (
 	"errors"
 	"fmt"
+	"strings"
 )
 
 var (
 	errNotFound         = errors.New("not found")
 	errInvalidArgument  = errors.New("invalid argument")
 	errInvalidOperation = errors.New("invalid operation")
 	errFailed           = errors.New("failed")
-	errUnsupported      = errors.New("unsupported")
 )
 
-func ErrUnsupported(what string, value interface{}) error {
-	return fmt.Errorf("%s=%v %w", what, value, errUnsupported)
+type pfcpifaceError struct {
+	message string
+	error   []error
 }
 
-func ErrNotFound(what string) error {
-	return fmt.Errorf("%s %w", what, errNotFound)
+func (e *pfcpifaceError) unwrap() string {
+	errMsg := strings.Builder{}
+	errMsg.WriteString("")
+
+	for i, e := range e.error {
+		errMsg.WriteString(fmt.Sprintf("\n- Error %v: %v", i, e))
+	}
+
+	return errMsg.String()
+}
+
+func (e *pfcpifaceError) Error() string {
+	return fmt.Sprintf("Message: %v. %v", e.message, e.unwrap())
+}
+
+func NewUnsupportedSourceIFaceError(iFaceType uint8, err ...error) *pfcpifaceError {
+	msg := fmt.Sprintf("Unsupported source interface type. Provided: %v", iFaceType)
+	return &pfcpifaceError{
+		message: msg,
+		error:   err,
+	}
+}
+
+func NewUnsupportedPrecedenceValue(value uint32, err ...error) *pfcpifaceError {
+	msg := fmt.Sprintf("Unsupported precedence greater than 65535. Provided: %v", value)
+	return &pfcpifaceError{
+		message: msg,
+		error:   err,
+	}
+}
+
+func NewNotFoundError(what string, err ...error) *pfcpifaceError {
```

Comment:
Added this to make a comparison. Is it better to have 'specialized' error methods? or would you go for more generic ones, like this?

Reply:
@osinstom @pudelkoM

Reply:
Not sure if an error for every single parameter is valuable. Do we need to differentiate on those in error handling code somewhere? Or is this about getting better error printing?

Reply:
I thought that having a better error structure could help (also for a better error printing). After a small discussion with @osinstom we agreed that it's not a priority right now 😃 Closing PR